back/app/services/cl2_data_listing_service.rb

```diff
@@ -19,30 +19,38 @@ def cl2_schema_models
     # After https://github.com/rails/rails/issues/37006.
     # It seems that we need to use Zeitwerk::Loader.eager_load_all
     # instead of Rails.application.eager_load!.
-    Zeitwerk::Loader.eager_load_all if Rails.env.development? || (File.basename($0) == 'rake')
+    Zeitwerk::Loader.eager_load_all if Rails.env.development? || (File.basename($PROGRAM_NAME) == 'rake')
     views = ActiveRecord::Base.connection.execute(
       "SELECT table_name FROM information_schema.tables WHERE table_type = 'VIEW'"
-      ).map{|r| r['table_name']}
+    ).map { |r| r['table_name'] }
     ActiveRecord::Base.descendants.select do |claz|
-      ![
+      [
         *ActiveRecord::Base.subclasses.map(&:name),
         PublicApi::ApiClient.name,
         Tenant.name
-      ].include? claz.name
+      ].exclude? claz.name
     end.select do |claz|
-      !views.include? claz.table_name
+      views.exclude? claz.table_name
     end
   end
 
-  def timestamp_attributes model_class
+  def timestamp_attributes(model_class)
     model_class.column_names.select do |column_name|
       column_name.ends_with? '_at'
     end
   end
 
-  def multiloc_attributes model_class
+  def multiloc_attributes(model_class)
     model_class.column_names.select do |column_name|
       column_name.ends_with? '_multiloc'
     end
   end
+
+  def html_multiloc_attributes(model_class)
+    multiloc_attributes(model_class).select do |atr|
+      model_class.validators.any? do |v|
+        v.attributes.include?(atr.to_sym) && v.options[:html]
```

Comment:
What is this line doing? Returning the multiloc column name as a symbol, but what is the v.options[:html] bit doing?

Would you mind giving me an example?

Reply:
I added an option in the multiloc validator, for example: `validates :description_multiloc, multiloc: { presence: false, html: true }`

The code you mention checks if the html: true part is specified. This allows us to list multiloc attributes that support HTML. When set to true, the validator will check if the HTML is valid.

Reply:
Makes sense. Thanks for explaining!